x/dsmr/node.go

```diff
@@ -264,5 +354,39 @@ func (n *Node[T]) Accept(ctx context.Context, block Block) error {
 		}
 	}
 
-	return n.storage.SetMin(block.Timestamp, chunkIDs)
+	if err := n.storage.SetMin(block.Timestamp, chunkIDs); err != nil {
+		return fmt.Errorf("failed to prune chunks: %w", err)
+	}
+
+	n.LastAccepted = block
+	return nil
+}
+
+type pChain struct {
+	validators []Validator
+}
+
+func (pChain) GetMinimumHeight(context.Context) (uint64, error) {
+	return 0, nil
+}
+
+func (pChain) GetCurrentHeight(context.Context) (uint64, error) {
+	return 0, nil
+}
+
+func (pChain) GetSubnetID(context.Context, ids.ID) (ids.ID, error) {
+	return ids.Empty, nil
+}
+
+func (p pChain) GetValidatorSet(context.Context, uint64, ids.ID) (map[ids.NodeID]*snowValidators.GetValidatorOutput, error) {
+	result := make(map[ids.NodeID]*snowValidators.GetValidatorOutput)
```

Comment:
it would be nice to create the map with the expected size. i.e. `len(p.validators)`